messaginginapp/src/test/java/io/customer/messaginginapp/ui/controller/InlineMessageViewControllerBehaviorTest.kt

```diff
@@ -0,0 +1,447 @@
+package io.customer.messaginginapp.ui.controller
+
+import io.customer.commontest.config.TestConfig
+import io.customer.commontest.config.testConfigurationDefault
+import io.customer.commontest.core.TestConstants
+import io.customer.commontest.extensions.assertNoInteractions
+import io.customer.commontest.extensions.attachToSDKComponent
+import io.customer.commontest.extensions.flushCoroutines
+import io.customer.commontest.extensions.random
+import io.customer.commontest.util.ScopeProviderStub
+import io.customer.messaginginapp.MessagingInAppModuleConfig
+import io.customer.messaginginapp.ModuleMessagingInApp
+import io.customer.messaginginapp.di.inAppMessagingManager
+import io.customer.messaginginapp.gist.GistEnvironment
+import io.customer.messaginginapp.gist.data.listeners.GistQueue
+import io.customer.messaginginapp.gist.data.model.Message
+import io.customer.messaginginapp.gist.data.model.engine.EngineWebConfiguration
+import io.customer.messaginginapp.gist.presentation.GistProvider
+import io.customer.messaginginapp.gist.presentation.GistSdk
+import io.customer.messaginginapp.gist.utilities.ElapsedTimer
+import io.customer.messaginginapp.state.InAppMessagingAction
+import io.customer.messaginginapp.state.InAppMessagingManager
+import io.customer.messaginginapp.testutils.core.JUnitTest
+import io.customer.messaginginapp.testutils.extension.createInAppMessage
+import io.customer.messaginginapp.ui.bridge.EngineWebViewDelegate
+import io.customer.messaginginapp.ui.bridge.InAppHostViewDelegate
+import io.customer.messaginginapp.ui.bridge.InAppPlatformDelegate
+import io.customer.messaginginapp.ui.bridge.InlineInAppMessageViewCallback
+import io.customer.sdk.core.di.SDKComponent
+import io.customer.sdk.core.util.ScopeProvider
+import io.customer.sdk.data.model.Region
+import io.mockk.MockKVerificationScope
+import io.mockk.clearMocks
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.spyk
+import io.mockk.verifyOrder
+import kotlin.math.roundToInt
+import org.amshove.kluent.shouldBeEqualTo
+import org.amshove.kluent.shouldBeNull
+import org.amshove.kluent.shouldNotBeNull
+import org.junit.jupiter.api.Test
+
+class InlineMessageViewControllerBehaviorTest : JUnitTest() {
```

Comment:
Can we add a test for error handling during message loading, for example, if an exception/error occurs during loading/display, how is it handled?

Reply:
Great catch. It's actually failing because there's a bug there 😞 I don't want the tests to keep failing, so I have put a ticket to include tests for it with the fix and working on it. Merging this PR for now and will create separate PR for the fix and test.